File: src/main/java/com/server/ordering/domain/member/Owner.java
package com.server.ordering.domain.member;

import com.server.ordering.domain.PhoneNumber;
import com.server.ordering.domain.Restaurant;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

import static javax.persistence.CascadeType.REMOVE;
import static javax.persistence.GenerationType.*;
import static lombok.AccessLevel.*;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
public class Owner extends MemberBase {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "owner_id")
    private Long id;

    @OneToOne(fetch = FetchType.LAZY, cascade = REMOVE, orphanRemoval = true)
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;

    public Owner(String signInId, String password, PhoneNumber phoneNumber) {
        this.signInId = signInId;
        this.password = password;
        this.phoneNumber = phoneNumber;
    }

    public void setRestaurant(Restaurant restaurant) {
        this.restaurant = restaurant;
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/member/Owner.kt
package com.byeoru.ordering_server.domain.member

import com.byeoru.ordering_server.domain.PhoneNumber
import com.byeoru.ordering_server.domain.Restaurant
import javax.persistence.*

@Entity
class Owner(signInId: String,
            password: String,
            phoneNumber: PhoneNumber) : MemberBase(signInId, password, phoneNumber) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "owner_id")
     var id: Long? = null
         protected set

    @OneToOne(fetch = FetchType.LAZY, cascade = [CascadeType.REMOVE], orphanRemoval = true)
    @JoinColumn(name = "restaurant_id")
     var restaurant: Restaurant? = null
         protected set

    fun registerRestaurant(restaurant: Restaurant?) {
        this.restaurant = restaurant
    }
}